src/libs/Network/index.js

```diff
@@ -61,17 +63,32 @@ function processPersistedRequestsQueue() {
 
     const tasks = _.map(persistedRequests, request => processRequest(request)
         .then((response) => {
-            if (response.jsonCode !== CONST.HTTP_STATUS_CODE.SUCCESS) {
-                throw new Error(`Persisted request failed due to jsonCode: ${response.jsonCode}`);
+            if (response.jsonCode === CONST.JSON_CODE.NOT_AUTHENTICATED) {
+                getLogger().info('Persisted optimistic request needs authentication');
+            } else {
+                getLogger().info('Persisted optimistic request returned a valid jsonCode. Not retrying.');
             }
-
+            onResponse(request, response);
```

Comment:
NAB: wouldn't be better to move `onResponse` to `processRequest` function?

Reply:
I'm gonna look into this in my next clean up PR. Basically there are two different `processRequest` happening here. One for regular requests run in the "main" queue and another for the persisted requests. Yes, I think there's probably a way to have just one handler for both. But I don't want to touch that yet.